tests/xdc/nexus_state_replication_test.go

```diff
@@ -494,6 +495,216 @@ func (s *NexusStateReplicationSuite) TestNexusCallbackReplicated() {
+	}, "workflow")
+	s.NoError(err)
+
+	pollResp := s.pollWorkflowTask(ctx, s.cluster1.FrontendClient(), ns)
+
+	// Complete first WFT with ScheduleNexusOperation and StartTimer(1s)
+	_, err = s.cluster1.FrontendClient().RespondWorkflowTaskCompleted(ctx, &workflowservice.RespondWorkflowTaskCompletedRequest{
+		Identity:  "test",
+		TaskToken: pollResp.TaskToken,
+		Commands: []*commandpb.Command{
+			{
+				CommandType: enumspb.COMMAND_TYPE_SCHEDULE_NEXUS_OPERATION,
+				Attributes: &commandpb.Command_ScheduleNexusOperationCommandAttributes{
+					ScheduleNexusOperationCommandAttributes: &commandpb.ScheduleNexusOperationCommandAttributes{
+						Endpoint:  endpointName,
+						Service:   "service",
+						Operation: "operation",
+					},
+				},
+			},
+			{
+				CommandType: enumspb.COMMAND_TYPE_START_TIMER,
+				Attributes: &commandpb.Command_StartTimerCommandAttributes{
+					StartTimerCommandAttributes: &commandpb.StartTimerCommandAttributes{
+						TimerId:            "timer-1",
+						StartToFireTimeout: durationpb.New(1 * time.Second),
+					},
+				},
+			},
+		},
+	})
+	s.NoError(err)
+
+	// Poll next WFT until TimerFired event is found
+	var secondPollResp *workflowservice.PollWorkflowTaskQueueResponse
+	var scheduledEventID int64
+	s.Eventually(func() bool {
+		secondPollResp, err = s.cluster1.FrontendClient().PollWorkflowTaskQueue(ctx, &workflowservice.PollWorkflowTaskQueueRequest{
+			Namespace: ns,
+			TaskQueue: &taskqueuepb.TaskQueue{Name: taskQueue, Kind: enumspb.TASK_QUEUE_KIND_NORMAL},
+			Identity:  "test",
+		})
+		if err != nil || secondPollResp == nil {
+			return false
+		}
+
+		var hasTimerFired bool
+		for _, e := range secondPollResp.History.Events {
+			if e.GetEventType() == enumspb.EVENT_TYPE_TIMER_FIRED {
+				hasTimerFired = true
+			}
+			if e.GetEventType() == enumspb.EVENT_TYPE_NEXUS_OPERATION_SCHEDULED {
+				scheduledEventID = e.GetEventId()
+			}
+		}
+		return hasTimerFired && scheduledEventID > 0
+	}, 10*time.Second, 200*time.Millisecond)
+
+	allowCompletion.Store(true)
+
+	s.Eventually(func() bool {
+		desc, err := sdkClient1.DescribeWorkflowExecution(ctx, run.GetID(), run.GetRunID())
+		s.NoError(err)
+		return len(desc.PendingNexusOperations) == 0
+	}, 10*time.Second, 200*time.Millisecond)
+
+	// Complete second WFT with RequestCancelNexusOperation command
+	_, err = s.cluster1.FrontendClient().RespondWorkflowTaskCompleted(ctx, &workflowservice.RespondWorkflowTaskCompletedRequest{
+		Identity:  "test",
+		TaskToken: secondPollResp.TaskToken,
+		Commands: []*commandpb.Command{
+			{
+				CommandType: enumspb.COMMAND_TYPE_REQUEST_CANCEL_NEXUS_OPERATION,
+				Attributes: &commandpb.Command_RequestCancelNexusOperationCommandAttributes{
+					RequestCancelNexusOperationCommandAttributes: &commandpb.RequestCancelNexusOperationCommandAttributes{
+						ScheduledEventId: scheduledEventID,
+					},
+				},
+			},
+		},
+	})
+
+	s.Error(err, "Cancel request should fail since operation is completed")
```

Comment:
This is the opposite of what we want, we should accept the cancel request since the operation completion is buffered.

Reply:
Fixed!